operator/redisfailover/service/constants.go

```diff
@@ -14,25 +14,27 @@ const (
 )
 
 const (
-	baseName                  = "rf"
-	sentinelName              = "s"
-	sentinelRoleName          = "sentinel"
-	sentinelConfigFileName    = "sentinel.conf"
-	sentinelNetworkPolicyName = "s-np"
-	redisConfigFileName       = "redis.conf"
-	redisName                 = "r"
-	redisNetworkPolicyName    = "r-np"
-	redisMasterName           = "rm"
-	redisSlaveName            = "rs"
-	redisShutdownName         = "r-s"
-	redisReadinessName        = "r-readiness"
-	redisRoleName             = "redis"
-	appLabel                  = "redis-failover"
-	hostnameTopologyKey       = "kubernetes.io/hostname"
+	baseName                   = "rf"
+	sentinelName               = "s"
+	sentinelRoleName           = "sentinel"
+	sentinelConfigFileName     = "sentinel.conf"
+	sentinelNetworkPolicyName  = "s-np"
+	redisConfigFileName        = "redis.conf"
+	redisName                  = "r"
+	redisNetworkPolicyName     = "r-np"
+	redisMasterName            = "rm"
+	redisSlaveName             = "rs"
+	redisShutdownName          = "r-s"
+	redisReadinessName         = "r-readiness"
+	redisRoleName              = "redis"
+	appLabel                   = "redis-failover"
+	hostnameTopologyKey        = "kubernetes.io/hostname"
+	redisHAProxySlaveRedisName = "rs-haproxy"
 )
 
 const (
 	redisRoleLabelKey    = "redisfailovers-role"
 	redisRoleLabelMaster = "master"
 	redisRoleLabelSlave  = "slave"
+	redisHARoleLabelKey  = "redishaproxy-role"
```

Comment:
In case of using label `redisfailovers-role:` we will face issue when redis services will point to the haproxy and not to the redis pods, this is because we will have similar selectors 
```
  selector:
    app.kubernetes.io/component: redis
    app.kubernetes.io/name: nitro-id-redis
    app.kubernetes.io/part-of: redis-failover
    redisfailovers-role: slave
```

As a solution, we have to change component of HAProxy, but this will require us to recreate HAProxy resources, because labels are [immutable](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates). So after upgrade version of redis-operator, we will have to manually delete HAProxy deployments from every app where it exists in every namespaces.
That is why I decided to use different label, I am not sure I want to mess with this manual step right now